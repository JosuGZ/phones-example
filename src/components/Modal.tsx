import * as React from 'react';

export interface ModalProps {
  title: JSX.Element | string;
  body: JSX.Element | string;
  onClose: () => void;
}

export class Modal extends React.Component<ModalProps, undefined> {

  shouldComponentUpdate(nextProps: ModalProps) {
    var updated = false;

    if (this.props.title !== nextProps.title) updated = true;
    if (this.props.body !== nextProps.body) updated = true;

    return updated;
  }

  render() {
    return (
      <div onClick={this.props.onClose} className="modal-container">
        <div className="modal-window">
          <div className="modal-window-header">
            <div className="title">{this.props.title}</div>
            <button onClick={this.props.onClose} className="close">X</button>
          </div>
          <div className="modal-window-body">
            {this.props.body}
          </div>
        </div>
      </div>
    );
  }

}
