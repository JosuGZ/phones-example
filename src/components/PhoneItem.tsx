import * as React from 'react';

export const PhoneItem = (props) => {
  return (
    <div className="phone-outer-box">
      <div className="phone">
        <div className="img-container">
          <img src={props.phone.image} />
        </div>
        <div className="name">{props.phone.name}</div>
      </div>
    </div>
  );
}
