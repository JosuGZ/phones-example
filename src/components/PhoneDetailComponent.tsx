import * as React from 'react';

import { Modal } from './Modal';

export interface PhoneDetailComponentProps {
  phone: any;
  onClose: () => void;
}

export class PhoneDetailComponent extends React.Component<PhoneDetailComponentProps, undefined> {

  shouldComponentUpdate(nextProps: PhoneDetailComponentProps) {
    var updated = false;

    if (this.props.phone !== nextProps.phone) updated = true;

    return updated;
  }

  render() {
    var props = this.props;
    var phone = props.phone;

    var title = phone.name;
    var body = (
      <div className="phone-detail-component">
        {
          phone.images && <div className="image-gallery">{
            phone.images.map((image, index) => {
              return <img key={index} src={image} />
            })
          }</div>
        }
        { phone.description && <p>{phone.description}</p> }
        { phone.processor && <p>Procesador: {phone.processor}</p> }
        { phone.ram && <p>Memoria RAM: {phone.ram}</p> }
        { phone.storage && <p>Almacenamiento: {phone.storage}</p> }
        { phone.screen && phone.screen.size && <p>Pantalla de {phone.screen.size}"</p> }
        { phone.screen && phone.screen.width && phone.screen.height && <p>Resoluci√≥n: {phone.screen.width}x{phone.screen.height}"</p> }
      </div>
    );

    return (
      <Modal title={title} body={body} onClose={props.onClose} />
    );
  }

}
