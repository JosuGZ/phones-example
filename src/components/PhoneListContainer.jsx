import * as React from 'react';
import { connect } from 'react-redux';

import { PhoneItem } from './PhoneItem';

// export interface PhoneListContainerProps {
//   phones: any[];
// }

export class PhoneListContainer extends React.Component {
  render() {
    return (
      <div className="phone-list-container">{
        this.props.phones.map((phone, index) => <PhoneItem key={index} phone={phone} />)
      }</div>
    );
  }
}
