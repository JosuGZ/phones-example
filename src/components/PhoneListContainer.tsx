import * as React from 'react';
import { connect } from 'react-redux';

import { PhoneItem } from './PhoneItem';

export interface PhoneListContainerProps {
  phones: any[];
}

export const PhoneListContainer = function(props: PhoneListContainerProps) {
  return (
    <div className="phone-list-container">{
      props.phones.map((phone, index) => <PhoneItem key={index} phone={phone} />)
    }</div>
  );
}
