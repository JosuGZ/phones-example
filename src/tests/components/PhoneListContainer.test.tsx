import * as React from 'react';
import { shallow } from 'enzyme';
import { PhoneListContainer } from '../../components/PhoneListContainer';

describe('Component PhoneListContainer', () => {
  const phones = [1, 2, 3, 4];
  const phoneListContainer = shallow(<PhoneListContainer phones={phones}/>);

  it('match snapshot', () => {
    expect(phoneListContainer).toMatchSnapshot();
  });

});
