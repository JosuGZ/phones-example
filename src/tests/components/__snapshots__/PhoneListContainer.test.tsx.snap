// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PhoneListContainer match snapshot 1`] = `
<div
  className="phone-list-container"
>
  <PhoneItem
    key="0"
    phone={1}
  />
  <PhoneItem
    key="1"
    phone={2}
  />
  <PhoneItem
    key="2"
    phone={3}
  />
  <PhoneItem
    key="3"
    phone={4}
  />
</div>
`;
